src/util/ajax.js

```diff
@@ -242,6 +242,12 @@ export const makeRequest = function(requestParameters: RequestParameters, callba
     //   some versions (see https://bugs.webkit.org/show_bug.cgi?id=174980#c2)
     // - Requests for resources with the file:// URI scheme don't work with the Fetch API either. In
     //   this case we unconditionally use XHR on the current thread since referrers don't matter.
+    if (/^custom:/.test(requestParameters.url) && isWorker() && self.worker && self.worker.actor) {
+        return self.worker.actor.send('getResource', requestParameters, callback);
+    }
+    if (/^custom:/.test(requestParameters.url) && !isWorker() && config.LOAD_TILES_FUNCTION != null) {
+        return config.LOAD_TILES_FUNCTION(requestParameters, callback);
```

Comment:
@HarelM I'm not an owner of this repo and don't have approval privileges, but I am very interested to see this change go into main. Anyway, this PR is failing in CircleCI. First issue is that the name of this function is causing the linter to fail. The linter doesn't like function names that start with an Uppercase letter, unless the function is a constructor. So LOAD_TILES_FUNCTION should be renamed to something like load_tiles_function or loadTilesFunction.

Reply:
Unfortunately, I have little faith it will be merged... :-( I talked about it in the relevant issue a long time ago, I asked if I should submit a PR and didn't got any answer. I opened this PR as part of Hacktoberfest just to make me feel better that I did everything I could in order to get this in :-)